src/Rings/mpolyquo-localizations.jl

```diff
@@ -1850,6 +1850,56 @@ function vector_space_dimension(R::MPolyQuoLocRing{<:Field, <:Any,<:Any, <:Any,
   return vector_space_dimension(quo(base_ring(R),ideal(base_ring(R),gens(LI)))[1])
 end
 
+@doc raw"""
+    monomial_basis(A::MPolyQuoLocRing{<:Field, <:Any, <:Any, <:Any, <:MPolyComplementOfKPointIdeal})
+    monomial_basis(L::MPolyLocRing{<:Field, <:Any, <:Any, <:Any, <:MPolyComplementOfKPointIdeal}, I::MPolyLocalizedIdeal)
+
+If, say, `A = L/I`, where `L` is a localization of multivariate polynomial ring over a field
+`K` at a point `p`, and `I` is an ideal of `L`, return a vector of monomials of `L`
+such that the residue classes of these monomials form a basis of `A` as a `K`-vector
+space. 
+!!! note 
+    The monomials are for readabilty in the varibles of the underlying polynomial ring of `L` and not in the variables of power series ring $K[[x_1-p_1,...,x_n-p_n]]$ in which `L` is embedded.
+!!! note
+    If `A` is not finite-dimensional as a `K`-vector space, an error is thrown. 
+# Examples
+```jldoctest
+julia> R, (x,y) = QQ["x","y"];
+
+julia> L,_ = localization(R, complement_of_point_ideal(R, [1,1]));
+
+julia> A,_ = quo(L, ideal(L, [(x-1)^2, (y-1)^2]));
+
+julia> A
+Localization
+  of quotient
+    of multivariate polynomial ring in 2 variables x, y
+      over rational field
+    by ideal (x^2 - 2*x + 1, y^2 - 2*y + 1)
+  at complement of maximal ideal of point (1, 1)
+
+julia> monomial_basis(A)
+4-element Vector{Oscar.MPolyLocRingElem{QQField, QQFieldElem, QQMPolyRing, QQMPolyRingElem, Oscar.MPolyComplementOfKPointIdeal{QQField, QQFieldElem, QQMPolyRing, QQMPolyRingElem}}}:
+ x*y
+ y
+ x
+ 1
+```
+"""
+function monomial_basis(L::MPolyLocRing{<:Field, <:Any, <:Any, <:Any, <:MPolyComplementOfKPointIdeal}, I::MPolyLocalizedIdeal)
+  base_ring(I) == L || error("ideal does not belong to the correct ring")
```

Comment:
Same question as above: why not
```suggestion
function monomial_basis(I::MPolyLocalizedIdeal)
  L = base_ring(I)
```
(though perhaps with a different name)

Reply:
See my remark above.

Reply:
This is not about the name, but about the type dispatch for the localization of the ring.

Reply:
The type of the ring is the parameter of the ideal, so it would be
```
blabla(I::MPolyLocalizedIdeal{MPolyLocRing{<:Field, <:Any, <:Any, <:Any, <:MPolyComplementOfKPointIdeal}})
```